android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/ScreenOrientationModule.java

```diff
@@ -55,47 +61,220 @@ public void onCatalystInstanceDestroy() {
+    }
+  }
+
+  @ReactMethod
+  public void getOrientationLockAsync(Promise promise) {
+    Activity activity = getCurrentActivity();
+
+    try {
+      int orientationAttr = activity.getRequestedOrientation();
+      OrientationLock orientationLock = orientationLockNativeToJS(orientationAttr);
+      promise.resolve(orientationLock.toString());
+    } catch (Exception e) {
+      promise.reject(ERR_SCREEN_ORIENTATION, "Could not get the current screen orientation lock", e);
+    }
+
+  }
+
+  @ReactMethod
+  public void getPlatformOrientationLockAsync(Promise promise) {
+    Activity activity = getCurrentActivity();
+    try {
+      promise.resolve(activity.getRequestedOrientation());
+    } catch (Exception e) {
+      promise.reject(ERR_SCREEN_ORIENTATION, "Could not get the current screen orientation platform lock", e);
+    }
+
+  }
+
+  @ReactMethod
+  public void supportsOrientationLockAsync(String orientationLockStr, Promise promise){
+    try {
+      // If we can get the native orientation value from the given string without throwing, we resolve with true
+      OrientationLock lockJS = OrientationLock.valueOf(orientationLockStr);
+      orientationLockJSToNative(lockJS);
+      promise.resolve(true);
+    } catch (Exception e) {
+      promise.resolve(false);
+    }
+  }
+
+  // https://stackoverflow.com/questions/10380989/how-do-i-get-the-current-orientation-activityinfo-screen-orientation-of-an-a
+  // Will not work in all cases as surface rotation is not standardized across android devices, but this is best effort
+  private Orientation getScreenOrientation(Activity activity) {
+    WindowManager windowManager = activity.getWindowManager();
+    int rotation = windowManager.getDefaultDisplay().getRotation();
+    DisplayMetrics dm = new DisplayMetrics();
+    windowManager.getDefaultDisplay().getMetrics(dm);
+    int width = dm.widthPixels;
+    int height = dm.heightPixels;
+    Orientation orientation;
+    // if the device's natural orientation is portrait:
+    if ((rotation == Surface.ROTATION_0
+        || rotation == Surface.ROTATION_180) && height > width ||
+        (rotation == Surface.ROTATION_90
+            || rotation == Surface.ROTATION_270) && width > height) {
+      switch (rotation) {
+        case Surface.ROTATION_0:
+          orientation = Orientation.PORTRAIT_UP;
+          break;
+        case Surface.ROTATION_90:
+          orientation = Orientation.LANDSCAPE_LEFT;
+          break;
+        case Surface.ROTATION_180:
+          orientation = Orientation.PORTRAIT_DOWN;
+          break;
+        case Surface.ROTATION_270:
+          orientation = Orientation.LANDSCAPE_RIGHT;
+          break;
+        default:
+          orientation = Orientation.UNKNOWN;
+          break;
+      }
+    }
+
+    // if the device's natural orientation is landscape or if the device
+    // is square:
+    else {
+      switch (rotation) {
+        case Surface.ROTATION_0:
+          orientation = Orientation.LANDSCAPE_LEFT;
```

Comment:
i would double check that these mappings are correct. it seems suspicious to me that they are not in the same order as the above case -- i.e. i would expect either LEFT, DOWN, RIGHT, UP, or RIGHT, UP, LEFT, DOWN.

Reply:
im not sure how to check this other than getting a real android tablet, so i will put it on my todo list and in the mean time, change it to LEFT,DOWN,RIGHT,UP since it makes the most sense with the code in the above case and the documentation https://stackoverflow.com/questions/5112118/how-to-detect-orientation-of-android-device#answer-5112135

Reply:
I'm not totally sure but I'd expect emulators to allow you to create a tablet and rotate it.